test/models/host_test.rb

```diff
@@ -61,5 +61,22 @@ class HostTest < ActiveSupport::TestCase
       assert_includes Host.search_for('compliant != true'), hosts(:one)
       assert_not_includes Host.search_for('compliant != false'), hosts(:one)
     end
+
+    should 'be able to filter by "has test results"' do
+      assert hosts(:one).test_results.present?
+      assert hosts(:two).test_results.empty?
+      assert_includes Host.search_for('has_test_results = true'), hosts(:one)
+      assert_not_includes(Host.search_for('has_test_results = true'),
+                          hosts(:two))
+      assert_includes Host.search_for('has_test_results = false'), hosts(:two)
+      assert_not_includes(Host.search_for('has_test_results = false'),
+                          hosts(:one))
+      assert_includes Host.search_for('has_test_results != true'), hosts(:two)
+      assert_not_includes(Host.search_for('has_test_results != true'),
+                          hosts(:one))
+      assert_includes Host.search_for('has_test_results != false'), hosts(:one)
```

Comment:
Oh! I'm a bit surprised these `!=` tests pass, since our code checks for `<>`.

Reply:
scoped_search automatically translates `!=` to `<>` as the operator

Reply:
Ah! nice, TIL